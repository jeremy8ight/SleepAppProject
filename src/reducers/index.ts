import {states} from '../states';
import {sleepInformationActions} from "../enums";
import {createStore} from "redux";

export const mainReducer = (state = states, action) => {
    switch (action.type) {
        case sleepInformationActions.storeFamilyMemberSleepInfo:
            return {...state, familyMemberIntervals: action.value.intervals}
        case sleepInformationActions.storeNameSelected:
            return {...state, nameSelected: action.value.name}
        default:
            return {...state};
    }
};

export const configureStore = (initialState = states) => {
    return createStore(mainReducer, initialState);
}
