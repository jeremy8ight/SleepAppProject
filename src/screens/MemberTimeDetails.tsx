import {View, Text, SafeAreaView, SectionList, TouchableOpacity} from "react-native";
import * as React from 'react';
import {useSelector} from "react-redux";
import {retrieveMemberNameSelector, retrieveMemberSleepInformationSelector} from "../selectors";
import {styles} from "../styles";
import {ListRow} from "../components/ListRow";
import {convertTime} from "../utils";
import {sleepApplicationRoute} from "../enums";

export const MemberTimeDetails = ({navigation}) => {
    const timeSleepData = useSelector(retrieveMemberSleepInformationSelector)
    const nameSelected = useSelector(retrieveMemberNameSelector)
    const rawTimeStorage = []
    timeSleepData.map((item) => {
        rawTimeStorage.push(item.ts)
    })
    const DATA = [
        {
            title: nameSelected,
            data: rawTimeStorage,
        },
    ];

    return (
        <SafeAreaView style={{backgroundColor: 'white'}}>
            <View style={styles.dashboard} testID={"memberTimeDetails:sectionList"}>
                <SectionList
                    stickySectionHeadersEnabled={true}
                    sections={DATA}
                    renderItem={({item}) => <ListRow  data={convertTime(item)} onPress={() => {
                        navigation.navigate(sleepApplicationRoute.memberData, {
                            time: item
                        })
                    }} showMemberName={false}/>}
                    renderSectionHeader={({section}) => (
                        <View style={styles.sectionHeaderViewStyle}>
                            <Text style={styles.loadingHeader}>{section.title}</Text>
                        </View>
                    )}
                    keyExtractor={item => item}
                />
            </View>
            <View style={styles.memberTimeDetailsButtons}>
                <TouchableOpacity
                    testID={'memberTimeDetails:goBack:button'}
                    style={styles.memberDetailBackBtn}
                    onPress={() => navigation.goBack()}>
                    <Text style={styles.loginText}>{"Go Back"}</Text>
                </TouchableOpacity>
                <TouchableOpacity
                    testID={'memberTimeDetails:signOut:button'}
                    style={styles.memberDetailBackBtn}
                    onPress={() => navigation.navigate(sleepApplicationRoute.signIn)}>
                    <Text style={styles.loginText}>{"Sign Out"}</Text>
                </TouchableOpacity>
            </View>
        </SafeAreaView>
    );
}
