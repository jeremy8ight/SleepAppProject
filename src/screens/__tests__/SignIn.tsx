import {configureStore} from '../../reducers';
import {Provider} from 'react-redux';
import {MemberTimeDetails} from '../MemberTimeDetails';
import {render} from '@testing-library/react-native';
import * as React from 'react';

describe('SignIn screen', () => {
  const mockNavigation = jest.fn();

  it('should return correctly', () => {
    const initialState = {
      familyMemberIntervals: [],
      nameSelected: 'jeremy',
    };
    const store = configureStore(initialState);
    const component = (
      <Provider store={store}>
        <MemberTimeDetails navigation={mockNavigation} />
      </Provider>
    );
    const result = render(component).toJSON();
    expect(result).toMatchSnapshot()
  });
});
