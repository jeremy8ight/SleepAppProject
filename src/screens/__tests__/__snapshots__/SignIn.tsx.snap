// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn screen should return correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "white",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": "100%",
      }
    }
    testID="memberTimeDetails:sectionList"
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "data": Array [],
            "title": "jeremy",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={
        Array [
          0,
        ]
      }
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "white",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 25,
                  "fontWeight": "800",
                }
              }
            >
              jeremy
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "bottom": 150,
        "flexDirection": "row",
        "height": "10%",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "purple",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": 100,
        }
      }
      testID="memberTimeDetails:goBack:button"
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Go Back
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "purple",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": 100,
        }
      }
      testID="memberTimeDetails:signOut:button"
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Sign Out
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
